submissions/api.py

```diff
@@ -637,13 +639,13 @@ def get_latest_score_for_submission(submission_uuid, read_replica=False):
         score = score_qs[0]
         if score.is_hidden():
             return None
-    except IndexError:
+    except (IndexError, Submission.DoesNotExist):
         return None
 
     return ScoreSerializer(score).data
 
 
-def reset_score(student_id, course_id, item_id):
+def reset_score(student_id, course_id, item_id, clear_state=False):
```

Comment:
Could you document the differences in behavior `clear_state` causes? It's not immediately clear because of the subtleties of how the submissions tables work.